qcelemental/info/cpu_info.py

```diff
@@ -0,0 +1,131 @@
+"""
+Contains metadata about Processors
+"""
+
+import difflib
+import re
+from enum import Enum
+from typing import Dict, Optional
+
+from pydantic import Field
+
+from ..models import ProtoModel
+
+
+class VendorEnum(str, Enum):
+    """Allowed quantum chemistry driver values.
+    """
+
+    amd = "amd"
+    intel = "intel"
+    nvidia = "nvidia"
+    arm = "arm"
+
+
+class InstructionSetEnum(Enum):
+    none = 0
+    sse = 1
+    avx = 2
+    avx2 = 3
+    avx512 = 4
+
+
+class ProcessorInfo(ProtoModel):
+    ncores: int = Field(..., description="The number of physical cores on the chip.")
+    nthreads: Optional[int] = Field(..., description="The maximum number of concurrent threads.")
+    base_clock: float = Field(..., description="The base clock frequency (GHz).")
+    boost_clock: Optional[float] = Field(..., description="The boost clock frequency (GHz).")
+    model: str = Field(..., description="The model number of the chip.")
+    family: str = Field(..., description="The family of the chip.")
+    launch_date: Optional[int] = Field(..., description="The launch year of the chip.")
+    target_use: str = Field(..., description="Target use case (Desktop, Server, etc).")
+    vendor: VendorEnum = Field(..., description="The vendor the chip is produced by.")
+    microarchitecture: Optional[str] = Field(..., description="The microarchitecture the chip follows.")
+    instructions: InstructionSetEnum = Field(..., description="The maximum vectorized instruction set available.")
+    type: str = Field(..., description="The type of chip (cpu, gpu, etc).")
+
+
+class ProcessorContext:
+    """Information about Process information.
+
+    Parameters
+    ----------
+    context : {'default'}
+        Origin of loaded data.
+
+    Attributes
+    ----------
+    functionals : Dict[str, ProcessorInfo]
+        dictionary of (lower-case) functional name to info
+
+    name : str
+        The name of the context ('default')
+    """
+
+    def __init__(self, context: str = "defualt"):
+        if context != "default":
+            raise KeyError(f"Context set as '{context}'," + " only contexts {'default'} are currently supported")
+
+        from .data import cpu_data_blob
+
+        self.processors: List[ProcessorInfo] = [ProcessorInfo(**data) for data in cpu_data_blob.data_blob]
+
+        self.index = {(x.vendor, self.process_names(x.model)): x for x in self.processors}
+
+        self.index_vendor = {k.name: {} for k in VendorEnum}
+        for proc in self.processors:
+            self.index_vendor[proc.vendor][self.process_names(proc.model)] = proc
+
+        self.name = context
+
+    def __str__(self) -> str:
+        return "ProcessorContext(context='{}')".format(self.name)
+
+    def process_names(self, name):
+        name = name.lower()
+        name = name.replace("(tm)", "").replace("(r)", "").replace("™", "")
+        name = name.replace("  ", " ").replace("\t", " ")
```

Comment:
`name = ' '.join(name.split())`

Reply:
I now use the same replace as in the `build_dft_info.py` to ensure they stay in sync.